sympy/physics/control/tests/test_lti.py

```diff
@@ -156,65 +155,122 @@ def test_TransferFunction_construction():
+
 
 def test_TransferFunction_functions():
     # explicitly cancel poles and zeros.
     tf0 = TransferFunction(s**5 + s**3 + s, s - s**2)
-    a = TransferFunction(-s**4 - s**2 - 1, s - 1)
-    assert tf0.cancel_poles_and_zeros() == a
+    a = TransferFunction(-(s**4 + s**2 + 1), s - 1)
+    assert tf0.simplify() == simplify(tf0) == a
 
     tf1 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5))
     b = TransferFunction(p + 3, p + 5)
-    assert tf1.cancel_poles_and_zeros() == b
+    assert tf1.simplify() == simplify(tf1) == b
+
+    # purely symbolic polynomials.
+    p1 = a1*s + a0
+    p2 = b2*s**2 + b1*s + b0
+    tf_ = TransferFunction(p1, p2, s)
+    expect = TransferFunction(2*s + 1, 5*s**2 + 4*s + 3)
+    assert tf_.evalf(subs={a0: 1, a1: 2, b0: 3, b1: 4, b2: 5}) == expect
 
     # negation of TF.
     tf2 = TransferFunction(s + 3, s**2 - s**3 + 9)
     tf3 = TransferFunction(-3*p + 3, 1 - p)
-    assert tf2.__neg__() == TransferFunction(-s - 3, s**2 - s**3 + 9)
-    assert tf3.__neg__() == TransferFunction(3*p - 3, 1 - p)
+    assert -tf2 == TransferFunction(-s - 3, s**2 - s**3 + 9)
+    assert -tf3 == TransferFunction(3*p - 3, 1 - p)
 
     # taking power of a TF.
     tf4 = TransferFunction(p + 4, p - 3)
     tf5 = TransferFunction(s**2 + 1, 1 - s)
-    result = TransferFunction(s**6 + 3*s**4 + 3*s**2 + 1, -s**3 + 3*s**2 - 3*s + 1)
-    assert tf4.pow(2) == TransferFunction(p**2 + 8*p + 16, p**2 - 6*p + 9)
-    assert tf5.pow(3) == result
+    expect2 = TransferFunction(s**6 + 3*s**4 + 3*s**2 + 1, -s**3 + 3*s**2 - 3*s + 1)
+    expect1 = TransferFunction(p**2 + 8*p + 16, p**2 - 6*p + 9)
+    assert tf4*tf4 == tf4**2 == pow(tf4, 2) == expect1
+    assert tf5*tf5*tf5 == tf5**3 == pow(tf5, 3) == expect2
+    assert tf4**-1 == pow(tf4, -1) == TransferFunction(p - 3, p + 4)
+
+    raises(ValueError, lambda: tf4**(s**2 + s - 1))
+    raises(ValueError, lambda: tf5**s)
+    raises(ValueError, lambda: tf4**tf5)
 
     # sympy's own functions.
     assert tf3.xreplace({p: s}) == TransferFunction(-3*s + 3, 1 - s)
     tf = TransferFunction(s - 1, s**2 - 2*s + 1)
+    tf_ = TransferFunction((s - 1)*(s + 3), s + 2)
     assert factor(tf) == TransferFunction(s - 1, (s - 1)**2)
+    assert tf.num.subs(s, 2) == tf.den.subs(s, 2) == 1
+    assert tf.subs(s, 2) == TransferFunction(1, 1)
+    assert expand(tf_) == TransferFunction(s**2 + 2*s - 3, s + 2)
 
 
-def test_TransferFunction_addition():
+def test_TransferFunction_addition_and_subtraction():
     tf1 = TransferFunction(s + 6, s - 5)
     tf2 = TransferFunction(s + 3, s + 1)
     tf3 = TransferFunction(s + 1, s**2 + s + 1)
     tf4 = TransferFunction(p, 2 - p)
-    assert tf1.add(tf2) == TransferFunction(2*s**2 + 5*s - 9, s**2 - 4*s - 5)
-    assert tf3.add(tf4) == \
+
+    # addition
+    assert tf1 + tf2 == TransferFunction(2*s**2 + 5*s - 9, s**2 - 4*s - 5)
+    assert tf3 + tf4 == \
         TransferFunction(p*s**2 + 2*s + 2, -p*s**2 - p*s - p**2 + 4*s + 2)
```

Comment:
The task of adding unit tests is almost over. What should be the variable for the resultant TF i.e. `tf3 + tf4`?

Reply:
That should raise and error.

Reply:
Basically every operation will have to check whether the polynomial variable is equivalent or an error is raised.